src/cdk/tree/tree.ts

```diff
@@ -0,0 +1,216 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {CollectionViewer, DataSource} from '@angular/cdk/collections';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ContentChildren,
+  Input,
+  IterableDiffers,
+  IterableDiffer,
+  IterableChangeRecord,
+  OnDestroy,
+  OnInit,
+  QueryList,
+  ViewChild,
+  ViewContainerRef,
+  ViewEncapsulation,
+} from '@angular/core';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+import {takeUntil} from 'rxjs/operator/takeUntil';
+import {Subject} from 'rxjs/Subject';
+import {CdkTreeNodeDef, CdkTreeNode, CdkTreeNodeOutletContext} from './node';
+import {TreeNodeOutlet} from './outlet';
+import {TreeControl} from './control/tree-control';
+import {
+  getTreeControlMissingError,
+  getTreeMissingMatchingNodeDefError,
+  getTreeMultipleDefaultNodeDefsError
+} from './tree-errors';
+
+
+/**
+ * CDK tree component that connects with a data source to retrieve data of type `T` and renders
+ * nodes with hierarchy. Updates the nodes when new data is provided by the data source.
+ */
+@Component({
+  moduleId: module.id,
+  selector: 'cdk-tree',
+  exportAs: 'cdkTree',
+  template: `<ng-container treeNodeOutlet></ng-container>`,
```

Comment:
Does this PR have the handling for nested tree nodes? It looks like they'd just be dropped as-is.

Reply:
Yes. The nested node handling will be done in `CdkNestedTreeNode` directive.